src/contracts/RichText/TextExtractorInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\FieldTypeRichText\RichText;
+
+use DOMDocument;
+
+interface TextExtractorInterface
+{
+    public function extractText(DOMDocument $node): string;
```

Comment:
Let's keep `$document` uniformly, as in both extractors:
```suggestion
    public function extractText(DOMDocument $documents): string;
```